lib/proj-parser.js

```diff
@@ -0,0 +1,77 @@
+var fs = require('fs');
+var path = require('path');
+var parseXML = require('xml2js').parseString;
+var debug = require('debug')('snyk');
+
+function determineDotnetVersion(rootDir) {
+  var csprojPath = findFile(rootDir, /.*\.csproj$/);
+  if (!csprojPath) {
+    throw new Error('.csproj file not found in ' + rootDir + '. ' +
+      'Make sure project was built before running `snyk test`');
+  }
+  debug('Checking .net framework version in .csproj file ' +
+    csprojPath);
+
+  var csprojContents = fs.readFileSync(csprojPath);
+
+  var frameworks = [];
+  parseXML(csprojContents, function (err, parsedCsprojContents) {
+    if (err) {
+      throw err;
+    }
+
+    var versionLoc = parsedCsprojContents.Project.PropertyGroup[0];
+
+    if (versionLoc.TargetFrameworkVersion) {
+      frameworks.push(toReadableVersion(versionLoc.TargetFrameworkVersion[0]));
+    } else if (versionLoc.TargetFramework) {
+      frameworks.push(toReadableVersion(versionLoc.TargetFramework[0]));
```

Comment:
what's happening in case toReadableVersion fails?

Reply:
[]

Reply:
see prev coment